src/scipp/io/csv.py

```diff
@@ -0,0 +1,141 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+
+"""Load CSV files.
+
+Note
+----
+CSV support requires `pandas <https://pandas.pydata.org/>`_ to be installed.
+You need to do this manually as it is not declared as a dependency of Scipp.
+You can use either ``pip install pandas`` or ``conda install -c conda-forge pandas``.
+
+
+CSV ('comma separated values') files store a simple table of data as a string.
+There are many different forms of this format.
+So check whether the loaded data is what you expect for your files.
+See ``scipp.io.csv.load_csv`` for examples.
+
+See Also
+--------
+pandas.read_csv:
+    More details on the underlying parser.
+"""
+
+from io import BytesIO, StringIO
+from os import PathLike
+from typing import Iterable, Optional, Union
+
+from ..compat.pandas_compat import HeadParserArg, from_pandas
+from ..core import Dataset
+
+
+# The typehint of filepath_or_buffer is less generic than in pd.read_table
+# because the definitions of protocols are private in pandas.
+def _load_dataframe(
+    filepath_or_buffer: Union[str, PathLike, StringIO, BytesIO], sep: str
+):
+    try:
+        import pandas as pd
+    except ImportError:
+        raise ImportError(
+            "Pandas is required to load CSV files but not install. "
+            "Install it with `pip install pandas` or "
+            "`conda install -c conda-forge pandas`."
+        ) from None
+    return pd.read_table(filepath_or_buffer, sep=sep)
+
+
+def load_csv(
+    filename: Union[str, PathLike, StringIO, BytesIO],
+    *,
+    sep: Optional[str] = '\t',
```

Comment:
Why is the default of `load_csv` a tab?

Reply:
Same as pandas. That's all I know

Reply:
Says `,` here: https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html. Maybe you are thinking of `load_table`?

Reply:
Yes